unittest/gunit/keyring_vault/vault_credentials_parser-t.cc

```diff
@@ -0,0 +1,288 @@
+#include <my_global.h>
+#include <gtest/gtest.h>
+#include "mock_logger.h"
+#include "vault_credentials_parser.h"
+#include <fstream>
+
+#if defined(HAVE_PSI_INTERFACE)
+namespace keyring
+{
+  PSI_memory_key key_memory_KEYRING = PSI_NOT_INSTRUMENTED;
+}
+#endif
+
+namespace keyring__vault_credentials_parser_unittest
+{
+  using namespace keyring;
+
+  using ::testing::StrEq;
+
+  class Vault_credentials_parser_test : public ::testing::Test
+  {
+  protected:
+    virtual void SetUp()
+    {
+      logger= new Mock_logger();
+    }
+
+    virtual void TearDown()
+    {
+      delete logger;
+      std::remove("./credentials");
+    }
+
+  protected:
+    ILogger *logger;
+  };
+
+  TEST_F(Vault_credentials_parser_test, ParseNotExistingFile)
+  {
+    Vault_credentials_parser vault_credentials_parser(logger);
+    std::string token;
+
+    EXPECT_CALL(*(reinterpret_cast<Mock_logger*>(logger)),
+      log(MY_ERROR_LEVEL, StrEq("Could not open file with credentials.")));
```

Comment:
Coupling in exact error message wording. Maybe create an array of such error messages in plugin with enum indexing?